pkg/kubelet/dockershim/network/hostport/hostport_manager.go

```diff
@@ -142,10 +143,11 @@ func (hm *hostportManager) Add(id string, podPortMapping *PodPortMapping, natInt
 			"-j", string(iptablesproxy.KubeMarkMasqChain))
 
 		// DNAT to the podIP:containerPort
+		hostPortBinding := net.JoinHostPort(podIP, strconv.Itoa(int(pm.ContainerPort)))
 		writeLine(natRules, "-A", string(chain),
 			"-m", "comment", "--comment", fmt.Sprintf(`"%s hostport %d"`, podFullName, pm.HostPort),
 			"-m", protocol, "-p", protocol,
-			"-j", "DNAT", fmt.Sprintf("--to-destination=%s:%d", podIP, pm.ContainerPort))
+			"-j", "DNAT", fmt.Sprintf("--to-destination=%s", hostPortBinding))
```

Comment:
`hostportManager.iptables` / `hostportSyncer.iptables` is specifically either an IPv4-only interface or an IPv6-only interface and maybe we should make sure it's the right one? (kubenet is the only user of this code in-tree, and it passes an IPv4-only interface, but I think it also only supports IPv4 pod IPs so that's ok)

Reply:
I' ve added checks to verify that the pod IP and the iptables version are the same